src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -48,5 +78,126 @@ public Optional<Credentials> getSchemaRegistryCredentials(String clusterId) {
     return Optional.ofNullable(credentials);
   }
 
-  // TODO: DIRECT connections need validation checks
+  @Override
+  public Future<ConnectionStatus> getConnectionStatus() {
+    return Future.join(
+        getKafkaConnectionStatus(),
+        getSchemaRegistryConnectionStatus()
+    ).map(cf -> {
+      var futures = cf.list();
+      var kafkaStatus = (KafkaClusterStatus) futures.get(0);
+      var srStatus = (SchemaRegistryStatus) futures.get(1);
+      return ConnectionStatusBuilder
+          .builder()
+          .kafkaCluster(kafkaStatus)
+          .schemaRegistry(srStatus)
+          .build();
+    });
+  }
+
+  protected Future<KafkaClusterStatus> getKafkaConnectionStatus() {
+    var kafkaConfig = spec.kafkaClusterConfig();
+    if (kafkaConfig == null) {
+      return Future.succeededFuture(
+          ConnectionStatusKafkaClusterStatusBuilder
+              .builder()
+              .state(ConnectedState.NONE)
+              .build()
+      );
+    }
+    // There is a Kafka configuration, so validate the connection by creating an AdminClient
+    // and describing the cluster.
+    try (var adminClient = createAdminClient(kafkaConfig)) {
+      var clusterDesc = adminClient.describeCluster();
+      var actualClusterId = clusterDesc.clusterId().get(5, TimeUnit.SECONDS);
```

Comment:
Should we make such timeouts configurable via the [application.yml](https://github.com/confluentinc/ide-sidecar/blob/main/src/main/resources/application.yml) or env vars? It would allow the extension to overwrite/change them.

Reply:
I don't anticipate the extension needing to change this, but I can add a configuration. How about a `ide-sidecar.connections.direct.timeout-seconds` property?